File: 프로그래머스/unrated/133499. 옹알이 （2）/옹알이 （2）.py
def solution(babbling):
    answer = 0
    words = ["aya", "ye", "woo", "ma" ]
    for bab in babbling:
        idx = 1
        tmp = bab[0]
        check = []
        while idx<len(bab):
            if tmp in words:
                check.append(tmp)
                bab=bab.replace(bab,bab[idx:])
                idx = 0
                tmp = bab[idx]
            else:
                tmp = tmp + bab[idx]
            idx+=1
        if tmp in words:
            check.append(tmp)
            if len(check)==1:
                answer+=1
            else:
                flag = True
                for i in range(len(check)-1):
                    if check[i] == check[i+1]:
                        flag = False
                        break
                if flag:
                    answer+=1

    return answer

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.java
class Solution {
    public int solution(String[] babbling) {
        int answer = 0;
        
        for (String bab : babbling) {
            // 연속 발음 체크
            if (bab.contains("ayaaya") || bab.contains("yeye") || 
                bab.contains("woowoo") || bab.contains("mama")) {
                continue;
            }
            
            // 발음 가능 단어 제거
            String temp = bab.replace("aya", " ")
                             .replace("ye", " ")
                             .replace("woo", " ")
                             .replace("ma", " ")
                             .replace(" ", "");
            
            // 모두 제거되었으면 발음 가능
            if (temp.isEmpty()) {
                answer++;
            }
        }
        
        return answer;
    }
}